### DISCUSSION TITLE
Must be called at the top of a `setup` function

### DISCUSSION THREAD

**Discussion by paulvonber** [nuxt-modules/i18n#3027] [Q&A]
Does anybody has clue why `const { t } = useI18n()` is not usable in the composable? is there any way to get translations in composable or not?

**Comment by BobbieGoede** [comment]
> Does anybody has clue why `const { t } = useI18n()` is not usable in the composable? is there any way to get translations in composable or not?

Can you provide a reproduction of your intended use case? It should be possible to do this in a composable as long as you call the composable in setup.

**Reply by Xenossolitarius** [reply]
I struggled with this too and never reported it. I mostly had this issue in pinia stores
Wrote a little helper to avoid this
```
export const useNuxtI18n = () => {
  const nuxtApp = useNuxtApp()

  return nuxtApp.$i18n
}
```
This should get me the global i18n instance.
Although I am really not sure if this is the correct usage and what problems can arise. 
It would be nice if someone could clarify
